packages/core/src/sensors/keyboard/KeyboardSensor.ts

```diff
@@ -39,15 +40,20 @@ export class KeyboardSensor implements SensorInstance {
 
     this.props = props;
     this.listeners = new Listeners(getOwnerDocument(target));
+    this.windowListeners = new Listeners(window);
     this.handleKeyDown = this.handleKeyDown.bind(this);
+    this.handleCancel = this.handleCancel.bind(this);
 
     this.attach();
   }
 
   private attach() {
     this.handleStart();
 
-    setTimeout(() => this.listeners.add('keydown', this.handleKeyDown));
+    setTimeout(() => {
+      this.listeners.add('keydown', this.handleKeyDown);
+      this.windowListeners.add('resize', this.handleCancel);
```

Comment:
Can instead just directly add listener to window without the `new Listeners(window)` instance.
```ts
window.addEventListener('resize', this.handleCancel);
```
@clauderic Whatever you think is best.